Sources/LiveKit/participant/LocalParticipant.swift

```diff
@@ -230,35 +232,47 @@ extension LocalParticipant {
         return set(source: .microphone, enabled: enabled)
     }
 
+    public func setScreen(enabled: Bool) -> Promise<LocalTrackPublication?> {
+        return set(source: .screenShareVideo, enabled: enabled)
+    }
+
     public func set(source: Track.Source, enabled: Bool, interceptor: VideoCaptureInterceptor? = nil) -> Promise<LocalTrackPublication?> {
         let publication = getTrackPublication(source: source)
-        if let publication = publication as? LocalTrackPublication {
+        if let publication = publication as? LocalTrackPublication,
+           let track = publication.track {
             // publication already exists
             if enabled {
                 publication.muted = false
+                track.start()
                 return Promise(publication)
             } else {
-                if source == .screenShareVideo {
-                    // screenshare cannot be muted
-                    return unpublish(publication: publication).then { _ in return Promise(nil) }
-                } else {
-                    publication.muted = true
-                    return Promise(nil)
-                }
+                publication.muted = true
+                track.stop()
+                return Promise(nil)
             }
         } else if enabled {
             // try to create a new track
-            do {
-                if source == .camera {
-                    let localTrack = try LocalVideoTrack.createCameraTrack(interceptor: interceptor)
-                    return publishVideoTrack(track: localTrack).then { publication in return publication }
-                }
-            } catch let error {
-                return Promise(error)
-            }
-            if source == .microphone {
+            if source == .camera {
+                let localTrack = LocalVideoTrack.createCameraTrack(interceptor: interceptor)
+                return publishVideoTrack(track: localTrack).then { publication in return publication }
+            } else if source == .microphone {
                 let localTrack = LocalAudioTrack.createTrack(name: "")
                 return publishAudioTrack(track: localTrack).then { publication in return publication }
+            } else if source == .screenShareVideo {
+
+                var localTrack: LocalVideoTrack?
+
+                #if !os(macOS)
+                if #available(macOS 11.0, *) {
```

Comment:
This looks a little odd? Tests if it's not macOS, but then checks if it's macOS 11.0 available? Is this supposed to be for iOS?

Reply:
thanks for finding this!! yea thats weird 😅